drivers/usb/udc/udc_common.c

```diff
@@ -22,7 +22,70 @@ LOG_MODULE_REGISTER(udc, CONFIG_UDC_DRIVER_LOG_LEVEL);
 
 static inline void udc_buf_destroy(struct net_buf *buf);
 
-NET_BUF_POOL_VAR_DEFINE(udc_ep_pool,
+/*
+ * Here we try to get DMA-safe buffers, but we lack a consistent source of
+ * information about data cache properties, such as line cache size, and a
+ * consistent source of information about what part of memory is DMA'able. For
+ * testing, we simply assume that all available memory is DMA'able and use
+ * hardcoded alignment and granularity.
+ */
+#if CONFIG_UDC_DWC2
+/*
+ * These are appropriate values for nRF54H20, but may be different on other
+ * platforms.
+ */
+#define BUF_ALIGN		32U
```

Comment:
Why not use `CONFIG_DCACHE_LINE_SIZE` instead? The DWC2 only requires alignment and granularity of 4 in Buffer DMA. The much larger alignment and granularity comes from the D-Cache.

Reply:
Also, should the align and granularity be available in some common header? It would make things simpler for UAC2 API users that do have to provide appropriate audio buffers.

Reply:
> Why not use `CONFIG_DCACHE_LINE_SIZE` instead? The DWC2 only requires alignment and granularity of 4 in Buffer DMA. The much larger alignment and granularity comes from the D-Cache.

That is for the the default case that DCACHE is enabled.

> Also, should the align and granularity be available in some common header? It would make things simpler for UAC2 API users that do have to provide appropriate audio buffers.

But it is not up to the UDC or USB subsystem to provide this information, we can have an interface to get it, but the source of this information should be part of the OS. This is the biggest problem, that we lack a consistent source of information about data cache properties and DMA'able areas at the OS level. And CONFIG_DCACHE_LINE_SIZE cannot be trusted because there is also d-cache-line-size' DT property, also there is Kconfig option DCACHE_LINE_SIZE_DETECT.

Reply:
I will try to come up with something more usable next week.